components/knowledge/Notion.tsx

```diff
@@ -0,0 +1,229 @@
+
+interface NotionFileModalProps {
+  isOpen: boolean;
+  onClose: () => void;
+  isSyncing: boolean;
+  setIsSyncing: React.Dispatch<React.SetStateAction<boolean>>;
+  notionConfigured: boolean;
+  setNotionConfigured: React.Dispatch<React.SetStateAction<boolean>>;
+  syncError: string;
+  setSyncError: React.Dispatch<React.SetStateAction<string>>;
+}
+
+export const NotionFileModal = ({
+  isOpen,
+  onClose,
+  notionConfigured,
+  setNotionConfigured,
+  isSyncing,
+  setIsSyncing,
+  syncError,
+  setSyncError,
+}: NotionFileModalProps) => {
+  const { droppedFiles, setDroppedFiles } = useContext(EditContext);
+  const [searchQuery, setSearchQuery] = useState<string>('');
+
+  const [notionFiles, setNotionFiles] = useState<
+    Map<string, { url: string; fileName: string }>
+  >(new Map());
+
+  const [selectedFile, setSelectedFile] = useState<string[]>(
+    Array.from(droppedFiles.keys())
+  );
+
+  useEffect(() => {
+    if (notionConfigured) return;
+
+    const setConfigured = async () => {
+      setNotionConfigured(await isNotionConfigured());
+    };
+    setConfigured();
+  }, [notionConfigured, isOpen]);
+
+  useEffect(() => {
+    if (!notionConfigured) return;
+
+    const setFiles = async () => {
+      const notionFiles = await getNotionFiles();
+      setNotionFiles(notionFiles);
+    };
+
+    setFiles();
+  }, [isOpen, notionConfigured]);
+
+  const onClickImport = async () => {
+    const files = await importFiles(selectedFile, 'notion');
+    setDroppedFiles((prev) => {
+      const newMap = new Map(prev);
+      for (const [key, file] of Array.from(newMap.entries())) {
+        if (file.type === 'notion') {
+          newMap.delete(key);
+        }
+      }
+      for (const [key, file] of Array.from(files.entries())) {
+        newMap.set(key, file);
+      }
+      return newMap;
+    });
+    onClose();
+  };
+
+  const syncNotion = async () => {
+    setIsSyncing(true);
+    try {
+      const isConfigured = await isNotionConfigured();
+      await runNotionSync(isConfigured);
+      setNotionConfigured(isConfigured);
+      setNotionFiles(await getNotionFiles());
+      setSyncError('');
+    } catch (e) {
+      setSyncError((e as Error).toString());
+    } finally {
+      setIsSyncing(false);
+    }
+  };
+
+  const handleSelectedFileChange = (selected: any) => {
+    if (selected === 'all') {
+      setSelectedFile(Array.from(notionFiles.keys()));
+    } else {
+      setSelectedFile(Array.from(selected as Set<string>));
+    }
+  };
+
+  return (
+    <Modal
+      size="4xl"
+      isOpen={isOpen}
+      onClose={onClose}
+      hideCloseButton={true}
+      scrollBehavior="inside"
+      className="min-h-[40vh] max-h-[80vh]"
+    >
+      <ModalContent>
+        <ModalHeader>
+          <div className="w-full flex items-center justify-between">
+            <div className="flex flex-row items-center">
+              <Image className="h-6 w-6" src="notion.svg" alt="Notion Icon" />
```

Comment:
nit and unimportant but:

```suggestion
              <Avatar size="sm" src="notion.svg" alt="Notion Icon" classNames={{base: "p-1.5 bg-white"}}/>
```
<details>

<summary>This makes it so that the logo isn't cut-off.</summary>

<img width="888" alt="Screenshot 2024-09-05 at 5 05 22 PM" src="https://github.com/user-attachments/assets/438b1826-159a-4213-bebb-b2f35aa0d366">